packages/yourloops/test/lib/team/hook.test.tsx

```diff
@@ -152,4 +179,293 @@ describe('Team hook', () => {
+    })
+  })
+
+  describe('getPatientRemoteMonitoringTeam', () => {
+    it('should throw an error if patient is not monitored', async () => {
+      await mountComponent()
+      expect(() => teamHook.getPatientRemoteMonitoringTeam(unknownPatient)).toThrowError('Cannot get patient remote monitoring team as patient is not remote monitored')
+    })
+
+    it('should throw an error if patient team could not be found', async () => {
+      const monitoredPatientWithUnknownTeam = createPatient('memberPatientAccepted1', undefined, undefined, undefined, {} as Monitoring)
+      await mountComponent()
+      expect(() => teamHook.getPatientRemoteMonitoringTeam(monitoredPatientWithUnknownTeam)).toThrowError('Could not find team to which patient is remote monitored')
+    })
+
+    it('should return the patient monitored team', async () => {
+      const expectedResult = {
+        teamId: 'team1Id',
+        status: UserInvitationStatus.accepted,
+        teamName: 'fakeTeamName'
+      } as PatientTeam
+      await mountComponent()
+      const team = teamHook.getPatientRemoteMonitoringTeam(monitoredPatient1)
+      expect(team).toEqual(expectedResult)
+    })
+  })
+
+  describe('getTeam', () => {
+    it('should return a team if exists or null', async () => {
+      await mountComponent()
+      let team = teamHook.getTeam('team1Id')
+      expect(team).toEqual(team1)
+      team = teamHook.getTeam('unknownId')
+      expect(team).toBeNull()
+    })
+  })
+
+  describe('getMedicalTeams', () => {
+    it('should return an array of medical teams', async () => {
+      await mountComponent()
+      const medicalTeams = teamHook.getMedicalTeams()
+      expect(medicalTeams).toEqual(teams)
+    })
+  })
+
+  describe('createTeam', () => {
+    it('should create a new team', async () => {
+      const initialTeamsLength = teams.length
+      expect(teams).toHaveLength(initialTeamsLength)
+      const newTeam = { id: 'newTeamId', email: 'fake@email.com', members: [] } as Team
+      jest.spyOn(TeamApi, 'createTeam').mockResolvedValueOnce({
+        id: 'newTeamId',
+        email: 'fake@email.com',
+        members: []
+      } as ITeam)
+      await mountComponent()
+      await act(async () => {
+        await teamHook.createTeam(newTeam)
+        expect(teams).toHaveLength(initialTeamsLength + 1)
+      })
+    })
+  })
+
+  describe('getPatient', () => {
+    it('should return a patient if existing in a team or null if not', async () => {
+      await mountComponent()
+      let patient = teamHook.getPatient('memberPatientAccepted1')
+      expect(patient).toBeTruthy()
+      patient = teamHook.getPatient('unknownId')
+      expect(patient).toBeFalsy()
+    })
+  })
+
+  describe('getPatients', () => {
+    it('should return an array of patients', async () => {
+      await mountComponent()
+      const patients = teamHook.getPatients()
+      expect(patients).toBeInstanceOf(Array)
+      expect(patients.length).toEqual(3)
```

Comment:
Again, `equal(3)` is not resilient. What is 3? If the test fails, the person fixing it might be tempting to juste replace 3 with the desired values.
To fix this, and having this, you can have:
`expect(patients.length).toEqual(allTeamMembers.filter(member => member.role === TeamMemberRole.patient).length)`

This is a drafty way of declaring the variable `allTeamMembers`:
```

  let allTeamMembers = []
  teams.forEach(team => {
    allTeamMembers.push(team.members)
  })
  allTeamMembers = allTeamMembers.flat()
  allTeamMembers = Array.from(new Set(allTeamMembers))
```
Feel free to improve it. This should be done at the top of the test file IMO (excpt if it need being updated by others tests)